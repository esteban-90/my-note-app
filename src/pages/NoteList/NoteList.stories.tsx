import type { ComponentStory, ComponentMeta } from '@storybook/react'
import { Layout } from '@/components'
import { NoteList } from './NoteList'

export default {
  title: 'Pages/Note List',
  component: NoteList,
  decorators: [
    (Story) => {
      return (
        <Layout>
          <Story />
        </Layout>
      )
    },
  ],
  parameters: {
    reactRouter: {
      routePath: '/',
    },
  },
} as ComponentMeta<typeof NoteList>

const Template: ComponentStory<typeof NoteList> = () => <NoteList />

export const NoteListExample = Template.bind({})
