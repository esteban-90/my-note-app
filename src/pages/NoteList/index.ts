export { NoteList as default } from './NoteList'
