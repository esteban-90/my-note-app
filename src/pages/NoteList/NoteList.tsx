import type { FC } from 'react'
import { Link } from 'react-router-dom'
import { FormattedMessage, useIntl } from 'react-intl'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { NoteItem, Button } from '@/components'
import { useNotes, useLanguage, useTheme } from '@/contexts'
import { Header, Title, Count, List } from '@styled/NoteList'

export const NoteList: FC = () => {
  const { notes } = useNotes()
  const { changeLanguage } = useLanguage()
  const { theme, changeTheme } = useTheme()

  const { formatMessage } = useIntl()
  const changeLanguageTitle = formatMessage({ id: 'note-list.change-language.title' })
  const dayModeTitle = formatMessage({ id: 'note-list.change-theme.day.title' })
  const nightModeTitle = formatMessage({ id: 'note-list.change-theme.night.title' })
  const newNoteTitle = formatMessage({ id: 'note-list.add-note.title' })

  return (
    <>
      <Header>
        <Title>
          <FontAwesomeIcon icon='book' /> <FormattedMessage id='note-list.heading' />
        </Title>
        <Count>{notes.length}</Count>
      </Header>
      <List>
        {notes.map((note) => (
          <NoteItem key={note.id} {...note} />
        ))}
      </List>
      <Button icon='language' title={changeLanguageTitle} onClick={changeLanguage} style={{ bottom: '10.5rem' }} />
      <Button
        icon={theme === 'night' ? 'sun' : 'moon'}
        title={theme === 'night' ? dayModeTitle : nightModeTitle}
        onClick={changeTheme}
        style={{ bottom: '6rem' }}
      />
      <Button icon='add' component={Link} to='/notes' title={newNoteTitle} role='button' style={{ bottom: '1.5rem' }} />
    </>
  )
}
