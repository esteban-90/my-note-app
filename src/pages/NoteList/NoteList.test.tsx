import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { getTitle, getDate, getBody } from '@/helpers'
import { mockNotes, saveNotes, removeNotes, saveTheme, removeTheme } from '@/mocks'
import { screen, customRender, userEvent, configure } from '@/tests'
import { NoteList } from './NoteList'

describe(NoteList.name, () => {
  let asFragment: () => DocumentFragment

  const UI = (
    <MemoryRouter>
      <Routes>
        <Route path='/'>
          <Route index element={<NoteList />} />
        </Route>
      </Routes>
    </MemoryRouter>
  )

  beforeAll(() => {
    configure({ throwSuggestions: true })
  })

  describe('when there are no notes:', () => {
    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('there should be a counter with value of "0"', () => {
      const counter = screen.getByText(/^0$/)
      expect(counter).toBeInTheDocument()
    })
  })

  describe('when there are more than zero notes:', () => {
    const counterValue = mockNotes.length

    beforeAll(saveNotes)

    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    afterAll(removeNotes)

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('they should be rendered', () => {
      mockNotes.forEach(({ content, createdAt }) => {
        const noteTitle = getTitle(content)
        const noteTitleElement = screen.getByRole('heading', { name: new RegExp(noteTitle, 'i') })
        expect(noteTitleElement).toBeInTheDocument()

        const noteDate = getDate(createdAt)
        const noteDateElement = screen.getByText(noteDate)
        expect(noteDateElement).toBeInTheDocument()

        const noteBody = getBody(content)
        const noteBodyElement = screen.getByText(noteBody, { trim: false })
        expect(noteBodyElement).toBeInTheDocument()
      })
    })

    it('each of them should be able to redirect to its detail page', () => {
      mockNotes.forEach(({ id, content }) => {
        const noteTitle = getTitle(content)
        const noteItemElement = screen.getByRole('link', { name: new RegExp(noteTitle, 'i') })
        expect(noteItemElement).toHaveAttribute('href', `/notes/${id}`)
      })
    })

    it(`there should be a counter with value of "${counterValue}"`, () => {
      const counter = screen.getByText(counterValue)
      expect(counter).not.toHaveTextContent('0')
      expect(counter).toBeInTheDocument()
    })
  })

  describe('when the language set is the one that comes by default (English):', () => {
    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    it('should match snapshot', () => {
      expect(asFragment()).toMatchSnapshot()
    })

    it('there should be a title with value of "My notes"', () => {
      const noteListHeading = screen.getByRole('heading', { name: /^my notes$/i })
      expect(noteListHeading).toBeInTheDocument()
    })

    describe('there should be a button with title of:', () => {
      it('"Change language"', () => {
        const changeLanguageButton = screen.getByRole('button', { name: /^change language$/i })
        expect(changeLanguageButton).toBeInTheDocument()
      })

      it('"New note", which should be able to redirect to note detail page if no id', () => {
        const addNoteButton = screen.getByRole('button', { name: /^new note$/i })
        expect(addNoteButton).toBeInTheDocument()
        expect(addNoteButton).toHaveAttribute('href', '/notes')
      })

      describe('and an additional one with title of:', () => {
        it('"Day mode" if the current theme mode is "night"', () => {
          const dayThemeButton = screen.getByRole('button', { name: /^day mode$/i })
          expect(dayThemeButton).toBeInTheDocument()
        })

        it('"Night mode" otherwise', async () => {
          const dayThemeButton = screen.getByRole('button', { name: /^day mode$/i })
          await userEvent.click(dayThemeButton)

          const nightThemeButton = await screen.findByRole('button', { name: /^night mode$/i })
          expect(nightThemeButton).toBeInTheDocument()
        })
      })
    })
  })

  describe('when the language is changed to Spanish:', () => {
    beforeAll(saveTheme)

    beforeEach(() => {
      ;({ asFragment } = customRender(UI))
    })

    afterAll(removeTheme)

    it('should match snapshot', async () => {
      const changeLanguageButton = screen.getByRole('button', { name: /change language/i })
      await userEvent.click(changeLanguageButton)

      expect(asFragment()).toMatchSnapshot()
    })

    it('there should be a title with value of "Mis notas"', () => {
      const noteListHeading = screen.getByRole('heading', { name: /^mis notas$/i })
      expect(noteListHeading).toBeInTheDocument()
    })

    describe('there should be a button with title of:', () => {
      it('"Cambiar idioma"', () => {
        const changeLanguageButton = screen.getByRole('button', { name: /^cambiar idioma$/i })
        expect(changeLanguageButton).toBeInTheDocument()
      })

      it('"Nueva nota", which should be able to redirect to note detail page if no id', () => {
        const addNoteButton = screen.getByRole('button', { name: /^nueva nota$/i })
        expect(addNoteButton).toBeInTheDocument()
        expect(addNoteButton).toHaveAttribute('href', '/notes')
      })

      describe('and an additional one with title of:', () => {
        it('"Modo día" if the current theme mode is "night"', () => {
          const dayThemeButton = screen.getByRole('button', { name: /^modo día$/i })
          expect(dayThemeButton).toBeInTheDocument()
        })

        it('"Modo noche" otherwise', async () => {
          const dayThemeButton = screen.getByRole('button', { name: /^modo día$/i })
          await userEvent.click(dayThemeButton)

          const nightThemeButton = await screen.findByRole('button', { name: /^modo noche$/i })
          expect(nightThemeButton).toBeInTheDocument()
        })
      })
    })
  })
})
