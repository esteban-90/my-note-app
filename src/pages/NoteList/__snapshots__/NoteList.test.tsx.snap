// Vitest Snapshot v1

exports[`NoteList > when the language set is Spanish: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-irlo6v"
  >
    <h2
      class="css-1kiihh0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-book "
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
       Mis notas
    </h2>
    <p
      class="css-1bgf1q4"
    >
      0
    </p>
  </div>
  <div
    class="css-kfrurf"
  />
  <button
    class="css-1gs37er"
    style="bottom: 10.5rem;"
    title="Cambiar idioma"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-language fa-inverse fa-xl "
      data-icon="language"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 128C0 92.7 28.7 64 64 64H256h48 16H576c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H320 304 256 64c-35.3 0-64-28.7-64-64V128zm320 0V384H576V128H320zM178.3 175.9c-3.2-7.2-10.4-11.9-18.3-11.9s-15.1 4.7-18.3 11.9l-64 144c-4.5 10.1 .1 21.9 10.2 26.4s21.9-.1 26.4-10.2l8.9-20.1h73.6l8.9 20.1c4.5 10.1 16.3 14.6 26.4 10.2s14.6-16.3 10.2-26.4l-64-144zM160 233.2L179 276H141l19-42.8zM448 164c11 0 20 9 20 20v4h44 16c11 0 20 9 20 20s-9 20-20 20h-2l-1.6 4.5c-8.9 24.4-22.4 46.6-39.6 65.4c.9 .6 1.8 1.1 2.7 1.6l18.9 11.3c9.5 5.7 12.5 18 6.9 27.4s-18 12.5-27.4 6.9l-18.9-11.3c-4.5-2.7-8.8-5.5-13.1-8.5c-10.6 7.5-21.9 14-34 19.4l-3.6 1.6c-10.1 4.5-21.9-.1-26.4-10.2s.1-21.9 10.2-26.4l3.6-1.6c6.4-2.9 12.6-6.1 18.5-9.8l-12.2-12.2c-7.8-7.8-7.8-20.5 0-28.3s20.5-7.8 28.3 0l14.6 14.6 .5 .5c12.4-13.1 22.5-28.3 29.8-45H448 376c-11 0-20-9-20-20s9-20 20-20h52v-4c0-11 9-20 20-20z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    class="css-1gs37er"
    style="bottom: 6rem;"
    title="Modo dÃ­a"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sun fa-inverse fa-xl "
      data-icon="sun"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
        fill="currentColor"
      />
    </svg>
  </button>
  <a
    class="css-1gs37er"
    href="/notes"
    role="button"
    style="bottom: 1.5rem;"
    title="Nueva nota"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-inverse fa-xl "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`NoteList > when the language set is the one that comes by default (English): > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-irlo6v"
  >
    <h2
      class="css-1kiihh0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-book "
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
       My notes
    </h2>
    <p
      class="css-1bgf1q4"
    >
      0
    </p>
  </div>
  <div
    class="css-kfrurf"
  />
  <button
    class="css-1gs37er"
    style="bottom: 10.5rem;"
    title="Change language"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-language fa-inverse fa-xl "
      data-icon="language"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 128C0 92.7 28.7 64 64 64H256h48 16H576c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H320 304 256 64c-35.3 0-64-28.7-64-64V128zm320 0V384H576V128H320zM178.3 175.9c-3.2-7.2-10.4-11.9-18.3-11.9s-15.1 4.7-18.3 11.9l-64 144c-4.5 10.1 .1 21.9 10.2 26.4s21.9-.1 26.4-10.2l8.9-20.1h73.6l8.9 20.1c4.5 10.1 16.3 14.6 26.4 10.2s14.6-16.3 10.2-26.4l-64-144zM160 233.2L179 276H141l19-42.8zM448 164c11 0 20 9 20 20v4h44 16c11 0 20 9 20 20s-9 20-20 20h-2l-1.6 4.5c-8.9 24.4-22.4 46.6-39.6 65.4c.9 .6 1.8 1.1 2.7 1.6l18.9 11.3c9.5 5.7 12.5 18 6.9 27.4s-18 12.5-27.4 6.9l-18.9-11.3c-4.5-2.7-8.8-5.5-13.1-8.5c-10.6 7.5-21.9 14-34 19.4l-3.6 1.6c-10.1 4.5-21.9-.1-26.4-10.2s.1-21.9 10.2-26.4l3.6-1.6c6.4-2.9 12.6-6.1 18.5-9.8l-12.2-12.2c-7.8-7.8-7.8-20.5 0-28.3s20.5-7.8 28.3 0l14.6 14.6 .5 .5c12.4-13.1 22.5-28.3 29.8-45H448 376c-11 0-20-9-20-20s9-20 20-20h52v-4c0-11 9-20 20-20z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    class="css-1gs37er"
    style="bottom: 6rem;"
    title="Day mode"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sun fa-inverse fa-xl "
      data-icon="sun"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
        fill="currentColor"
      />
    </svg>
  </button>
  <a
    class="css-1gs37er"
    href="/notes"
    role="button"
    style="bottom: 1.5rem;"
    title="New note"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-inverse fa-xl "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`NoteList > when there are more than zero notes: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-irlo6v"
  >
    <h2
      class="css-1kiihh0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-book "
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
       My notes
    </h2>
    <p
      class="css-1bgf1q4"
    >
      5
    </p>
  </div>
  <div
    class="css-kfrurf"
  >
    <a
      data-testid="note-99lbu3vwlamz31ru"
      href="/notes/note-99lbu3vwlamz31ru"
      role="link"
    >
      <div
        class="css-14m53i8"
      >
        <h3>
          Hello test 1
        </h3>
        <p>
          <span>
            December 17, 1995 at 3:24:00 AM
          </span>
          <span>
             Testing 1
          </span>
        </p>
      </div>
    </a>
    <a
      data-testid="note-99lbu3vwlamz31rv"
      href="/notes/note-99lbu3vwlamz31rv"
      role="link"
    >
      <div
        class="css-14m53i8"
      >
        <h3>
          Hello test 2
        </h3>
        <p>
          <span>
            December 17, 1995 at 3:24:01 AM
          </span>
          <span>
             Testing 2
          </span>
        </p>
      </div>
    </a>
    <a
      data-testid="note-99lbu3vwlamz31rw"
      href="/notes/note-99lbu3vwlamz31rw"
      role="link"
    >
      <div
        class="css-14m53i8"
      >
        <h3>
          Hello test 3
        </h3>
        <p>
          <span>
            December 17, 1995 at 3:24:02 AM
          </span>
          <span>
             Testing 3
          </span>
        </p>
      </div>
    </a>
    <a
      data-testid="note-99lbu3vwlamz31rx"
      href="/notes/note-99lbu3vwlamz31rx"
      role="link"
    >
      <div
        class="css-14m53i8"
      >
        <h3>
          Hello test 4
        </h3>
        <p>
          <span>
            December 17, 1995 at 3:24:03 AM
          </span>
          <span>
             Testing 4
          </span>
        </p>
      </div>
    </a>
    <a
      data-testid="note-99lbu3vwlamz31ry"
      href="/notes/note-99lbu3vwlamz31ry"
      role="link"
    >
      <div
        class="css-14m53i8"
      >
        <h3>
          Hello test 5
        </h3>
        <p>
          <span>
            December 17, 1995 at 3:24:04 AM
          </span>
          <span>
             Testing 5
          </span>
        </p>
      </div>
    </a>
  </div>
  <button
    class="css-1gs37er"
    style="bottom: 10.5rem;"
    title="Change language"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-language fa-inverse fa-xl "
      data-icon="language"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 128C0 92.7 28.7 64 64 64H256h48 16H576c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H320 304 256 64c-35.3 0-64-28.7-64-64V128zm320 0V384H576V128H320zM178.3 175.9c-3.2-7.2-10.4-11.9-18.3-11.9s-15.1 4.7-18.3 11.9l-64 144c-4.5 10.1 .1 21.9 10.2 26.4s21.9-.1 26.4-10.2l8.9-20.1h73.6l8.9 20.1c4.5 10.1 16.3 14.6 26.4 10.2s14.6-16.3 10.2-26.4l-64-144zM160 233.2L179 276H141l19-42.8zM448 164c11 0 20 9 20 20v4h44 16c11 0 20 9 20 20s-9 20-20 20h-2l-1.6 4.5c-8.9 24.4-22.4 46.6-39.6 65.4c.9 .6 1.8 1.1 2.7 1.6l18.9 11.3c9.5 5.7 12.5 18 6.9 27.4s-18 12.5-27.4 6.9l-18.9-11.3c-4.5-2.7-8.8-5.5-13.1-8.5c-10.6 7.5-21.9 14-34 19.4l-3.6 1.6c-10.1 4.5-21.9-.1-26.4-10.2s.1-21.9 10.2-26.4l3.6-1.6c6.4-2.9 12.6-6.1 18.5-9.8l-12.2-12.2c-7.8-7.8-7.8-20.5 0-28.3s20.5-7.8 28.3 0l14.6 14.6 .5 .5c12.4-13.1 22.5-28.3 29.8-45H448 376c-11 0-20-9-20-20s9-20 20-20h52v-4c0-11 9-20 20-20z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    class="css-1gs37er"
    style="bottom: 6rem;"
    title="Day mode"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sun fa-inverse fa-xl "
      data-icon="sun"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
        fill="currentColor"
      />
    </svg>
  </button>
  <a
    class="css-1gs37er"
    href="/notes"
    role="button"
    style="bottom: 1.5rem;"
    title="New note"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-inverse fa-xl "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`NoteList > when there are no notes: > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-irlo6v"
  >
    <h2
      class="css-1kiihh0"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-book "
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96V416c0 53 43 96 96 96H384h32c17.7 0 32-14.3 32-32s-14.3-32-32-32V384c17.7 0 32-14.3 32-32V32c0-17.7-14.3-32-32-32H384 96zm0 384H352v64H96c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16zm16 48H336c8.8 0 16 7.2 16 16s-7.2 16-16 16H144c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
       My notes
    </h2>
    <p
      class="css-1bgf1q4"
    >
      0
    </p>
  </div>
  <div
    class="css-kfrurf"
  />
  <button
    class="css-1gs37er"
    style="bottom: 10.5rem;"
    title="Change language"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-language fa-inverse fa-xl "
      data-icon="language"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 128C0 92.7 28.7 64 64 64H256h48 16H576c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H320 304 256 64c-35.3 0-64-28.7-64-64V128zm320 0V384H576V128H320zM178.3 175.9c-3.2-7.2-10.4-11.9-18.3-11.9s-15.1 4.7-18.3 11.9l-64 144c-4.5 10.1 .1 21.9 10.2 26.4s21.9-.1 26.4-10.2l8.9-20.1h73.6l8.9 20.1c4.5 10.1 16.3 14.6 26.4 10.2s14.6-16.3 10.2-26.4l-64-144zM160 233.2L179 276H141l19-42.8zM448 164c11 0 20 9 20 20v4h44 16c11 0 20 9 20 20s-9 20-20 20h-2l-1.6 4.5c-8.9 24.4-22.4 46.6-39.6 65.4c.9 .6 1.8 1.1 2.7 1.6l18.9 11.3c9.5 5.7 12.5 18 6.9 27.4s-18 12.5-27.4 6.9l-18.9-11.3c-4.5-2.7-8.8-5.5-13.1-8.5c-10.6 7.5-21.9 14-34 19.4l-3.6 1.6c-10.1 4.5-21.9-.1-26.4-10.2s.1-21.9 10.2-26.4l3.6-1.6c6.4-2.9 12.6-6.1 18.5-9.8l-12.2-12.2c-7.8-7.8-7.8-20.5 0-28.3s20.5-7.8 28.3 0l14.6 14.6 .5 .5c12.4-13.1 22.5-28.3 29.8-45H448 376c-11 0-20-9-20-20s9-20 20-20h52v-4c0-11 9-20 20-20z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    class="css-1gs37er"
    style="bottom: 6rem;"
    title="Day mode"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-sun fa-inverse fa-xl "
      data-icon="sun"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
        fill="currentColor"
      />
    </svg>
  </button>
  <a
    class="css-1gs37er"
    href="/notes"
    role="button"
    style="bottom: 1.5rem;"
    title="New note"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-plus fa-inverse fa-xl "
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;
